### PR TITLE
feat: add support for the Pine64 Rock64

### PR DESCRIPTION
**Pull request by andrewrynhard**
This adds support for the Rock64.

Signed-off-by: Andrew Rynhard <andrew@rynhard.io>

### CODE REVIEW

File: internal/app/machined/pkg/runtime/v1alpha1/board/rock64/rock64.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	}
+
+	return nil
+}
+
+// KernelArgs implements the runtime.Board.
+func (r *Rock64) KernelArgs() procfs.Parameters {
+	return []*procfs.Parameter{
+		procfs.NewParameter("console").Append("ttyS2,115200n8"),
+		procfs.NewParameter("console").Append("uart8250,mmio32,0xff130000"),
```
Comment (andrewrynhard) [inline internal/app/machined/pkg/runtime/v1alpha1/board/rock64/rock64.go:94]: This seems to cause double printing.

File: internal/app/machined/pkg/runtime/v1alpha1/board/rock64/rock64.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	}
+}
+
+// PartitionOptions implements the runtime.Board.
+func (r *Rock64) PartitionOptions() *runtime.PartitionOptions {
+	return &runtime.PartitionOptions{PartitionsOffset: 2048 * 9}
```
Comment (andrewrynhard) [inline internal/app/machined/pkg/runtime/v1alpha1/board/rock64/rock64.go:100]: do we need to give a little bit of buffer space? the spl is close to 9mb

**Review by rsmitty — APPROVED** [review]

**Pull request merged**